File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
	attr_reader :num

	def initialize(number)
		@number = number
	end

	def square_of_sums
		@number.downto(1).reduce(:+)**2
	end

	def sum_of_squares
		@number.downto(1).map{|x| x**2}.reduce(:+)
	end

	def difference
		square_of_sums - sum_of_squares
	end
end

File: python/difference-of-squares/difference_of_squares.py
import functools as f

def difference(num):
  return square_of_sum(num) - sum_of_squares(num)

def square_of_sum(num):
  return f.reduce(lambda x,y: x+y, list(range(num+1)))**2

def sum_of_squares(num):
  return f.reduce(lambda x,y: x+y, map(lambda x: x**2, list(range(num+1))))
